controllers/solrprometheusexporter_controller_test.go

```diff
@@ -656,10 +657,14 @@ func TestMetricsReconcileWithUserProvidedConfig(t *testing.T) {
 	updatedConfigXml := "<config>updated by user</config>"
 	updateUserProvidedConfigMap(testClient, g, userProvidedConfigMapNN, map[string]string{util.PrometheusExporterConfigMapKey: updatedConfigXml})
 
-	// reconcile should happen again
+	// reconcile should happen again 2x
 	g.Eventually(requests, timeout).Should(gomega.Receive(gomega.Equal(expectedMetricsRequest)))
+	g.Eventually(requests, timeout).Should(gomega.Receive(gomega.Equal(expectedMetricsRequest)))
+
```

Comment:
In the [solrxml test](https://github.com/apache/lucene-solr-operator/blob/main/controllers/solrcloud_controller_test.go#L772), we added a 250 ms sleep to make sure it passed consistently. Might make sense to have here as well.